drivers/audio/codec_shell.c

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright (c) 2023 Centralp
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/shell/shell.h>
+#include <zephyr/audio/codec.h>
+
+#define CODEC_START_HELP                                                                           \
+	"Start output audio playback. Syntax:\n"                                                   \
+	"<device>"
+
+#define CODEC_STOP_HELP                                                                            \
+	"Stop output audio playback. Syntax:\n"                                                    \
+	"<device>"
+
+#define CODEC_SET_PROP_HELP                                                                        \
+	"Set a codec property. Syntax:\n"                                                          \
+	"<device> <property> <channel> <value>"
+
+#define CODEC_APPLY_PROP_HELP                                                                      \
+	"Apply any cached properties. Syntax:\n"                                                   \
+	"<device>"
+
+static const char *const codec_property_name[] = {
+	[AUDIO_PROPERTY_OUTPUT_VOLUME] = "volume",
+	[AUDIO_PROPERTY_OUTPUT_MUTE] = "mute",
+};
+
+static const char *const codec_channel_name[] = {
+	[AUDIO_CHANNEL_FRONT_LEFT] = "front_left",
+	[AUDIO_CHANNEL_FRONT_RIGHT] = "front_right",
+	[AUDIO_CHANNEL_LFE] = "lfe",
+	[AUDIO_CHANNEL_FRONT_CENTER] = "front_center",
+	[AUDIO_CHANNEL_REAR_LEFT] = "rear_left",
+	[AUDIO_CHANNEL_REAR_RIGHT] = "rear_right",
+	[AUDIO_CHANNEL_REAR_CENTER] = "rear_center",
+	[AUDIO_CHANNEL_SIDE_LEFT] = "side_left",
+	[AUDIO_CHANNEL_SIDE_RIGHT] = "side_right",
+	[AUDIO_CHANNEL_ALL] = "all",
+};
+
+struct args_index {
+	uint8_t device;
+	uint8_t property;
+	uint8_t channel;
+	uint8_t value;
+};
+
+static const struct args_index args_indx = {
+	.device = 1,
+	.property = 2,
+	.channel = 3,
+	.value = 4,
+};
+
```

Comment:
I think we should avoid magic numbers. Something like:

```
#define SHELL_ARG_DEVICE_INDEX 1
#define SHELL_ARG_PROPERTY_INDEX 2
```

Reply:
The same trick is used in several (6) shell files:

https://github.com/zephyrproject-rtos/zephyr/blob/6d67a56d5b5ddb3f744eb844e6b4f7d033be7a94/drivers/eeprom/eeprom_shell.c#L17-L31

https://github.com/zephyrproject-rtos/zephyr/blob/6d67a56d5b5ddb3f744eb844e6b4f7d033be7a94/drivers/gpio/gpio_shell.c#L24-L36

...

I propose you to leave it as it